pkg/lspext/cacheext.go

```diff
@@ -0,0 +1,23 @@
+package lspext
+
+// CacheGetParams is the input for 'cache/xget'. The response is a
+// 'CacheItem'. This cache is global to a language server. IE it is shared
+// amongst workspaces, but not amongst different language server types.
+type CacheGetParams struct {
+	Key string `json:"key"`
+}
+
+// CacheItem is the response for 'cache/xget'.
+type CacheItem struct {
+	// Ok is true if the item was successfuly retrieved from the cache.
+	Ok bool `json:"ok"`
```

Comment:
Sure we can do that as well. I did it this way for three reasons:
* Explicit.
* Avoid nulls.
* Avoid empty values (eg `""`) having a special meaning.

For example in some languages the JSON parser makes it difficult to distinguish between a value that isn't present vs empty (languages that generally attempt to deal with null/undefined much better than dynamic languages do)